package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,9 +150,34 @@ def parse(self, **kwargs):
         resids = []
         resnames = []
         charges = []
-
+        set_charge = True
+        for line in sections['molecule']:
+            if line.strip().lower() == 'no_charges':
+                set_charge = False
+                break
```

Comment:
From the [MOL2 file standard](https://zhanggroup.org/DockRMSD/mol2.pdf), the 4th line of the `@<TRIPOS>MOLECULE` section specifies the charge. You should be able to condense this in a single line. All supported charge types are uppercase.

Reply:
@RMeli Thank you for this one. It's very reasonable and also saves loop complexity.